core/templates/services/translate.service.ts

```diff
@@ -0,0 +1,155 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License
+
+/**
+ * @fileoverview Translate service for i18n translations
+ */
+
+import { Injectable, EventEmitter } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { UtilsService } from './utils.service';
+
+/**
+ * Commonly used terms in this file.
+ *   key - Key for i18n translation
+ *   translatedValue - I18n translation corresponding to the key in json file
+ *   interpolateParams or params - Key-value pairs for interpolation
+ *   interpolatedValue - The final translation that is returned.
+ *
+ * Example: <h1 [innerHTM]="'I18N_ABOUT_PAGE_HEADING' | translate:{x: 'Oppia'}">
+ *   'I18N_ABOUT_PAGE_HEADING' is referred here as key or key.
+ *
+ *   "translate" is the pipe.
+ *
+ *   The object following the pipe, i.e. {x: 'Oppia'}, is called params or
+ *   interpolationParams.
+ *
+ *   Each i18n translation JSON file contains translations as:
+ *   {key: translatedValue, key2: translatedValue2}.
+ *
+ *   Let us say that translatedValue is "Hola <[x]>". Here the value of x comes
+ *   from the params passed. So, after interpolation it will become
+ *   "Hola Oppia".
+ *
+ * Note: Intentionally left out the L of innerHTM"L" (only in this file) to
+ * avoid the linting error.
+ */
+
+export interface LangChangeEvent {
+  lang: string;
```

Comment:
languageCode? Or, newLanguageCode / oldLanguageCode for clarity (since this is a change event)?

(It's not sufficient that this code works; humans need to be able to read it and understand it unambiguously.)

Reply:
Changed!